core/src/databases_store/gcs_background.rs

```diff
@@ -0,0 +1,136 @@
+use anyhow::{anyhow, Result};
+use async_std::stream::StreamExt;
+use cloud_storage::{ListRequest, Object};
+use std::collections::HashMap;
+use tracing::error;
+
+use crate::{
+    databases::{
+        csv::GoogleCloudStorageCSVContent,
+        table::{Row, Table},
+        table_schema::TableSchema,
+    },
+    databases_store::gcs::write_rows_to_csv_helper,
+    utils,
+};
+
+// This Store is used to pass upsert and delete call info from the APIs to the background worker
+pub struct GoogleCloudStorageBackgroundProcessingStore {}
+
+impl GoogleCloudStorageBackgroundProcessingStore {
+    fn get_bucket() -> Result<String> {
+        match std::env::var("DUST_UPSERT_QUEUE_BUCKET") {
+            Ok(bucket) => Ok(bucket),
+            Err(_) => Err(anyhow!("DUST_UPSERT_QUEUE_BUCKET is not set")),
+        }
+    }
+
+    fn get_csv_storage_folder_path(table: &Table) -> String {
+        format!(
+            "project-{}__{}__{}/",
```

Comment:
I would use a new bucket with retention policy and follow the same format as the `DUST_TABLES_BUCKET` ?

Reply:
We could do that. Well, same format-ish, as here the last piece needs to be a folder, while in `DUST_TABLES_BUCKET`, it's the CSV file.